velox/exec/HashProbe.cpp

```diff
@@ -874,11 +875,6 @@ RowVectorPtr HashProbe::getOutput() {
 
     numOut = evalFilter(numOut);
     if (!numOut) {
-      // The filter was false on all rows.
-      if (isLeftSemiOrAntiJoinNoFilter) {
```

Comment:
This is a fix for the other join type as well? Thanks!

Reply:
This is dead code. isLeftSemiOrAntiJoinNoFilter is true only if there is no filter. Hence, numOut before evalFilter is the same as numOut after. We already checked if numOut is zero before the filter, hence, this branch is not reachable.